org.eclipse.debug.tests/src/org/eclipse/debug/tests/console/TextConsoleViewerTest.java

```diff
@@ -303,4 +305,49 @@ private static void checkOverlapping(List<StyleRange> styles) {
 			lastEnd = s.start + s.length;
 		}
 	}
+
+	/**
+	 * Test that findPosition returns only all overlapping positions
+	 */
+	@Test
+	public void testVisitOverlappingPositions() throws Throwable {
+
+		try {
+			final Method method = TextConsoleViewer.class.getDeclaredMethod("visitOverlappingPositions", int.class, int.class, Position[].class, Consumer.class);
```

Comment:
Just wondering if rather than using reflection you can make the method public, but not part of API by using `@noreference`?

I know there are other examples in this test code that do similar, but by using reflection here it means longer term maintenance is harder because things like Refactoring and Call Hierarchy won't show this connection.